core/trino-spi/src/main/java/io/trino/spi/Page.java

```diff
@@ -315,6 +315,20 @@ public Page getLoadedPage(int... columns)
         return wrapBlocksWithoutCopy(positionCount, blocks);
     }
 
+    public Page getLoadedPage(int[] columns, int[] eagerlyLoadedColumnIndexes)
```

Comment:
nit: `eagerlyLoadedColumnIndexes` -> `eagerlyLoadedColumns`

Reply:
I'm calling this `eagerlyLoadedColumnIndexes` rather than `eagerlyLoadedColumns` as these are indices in `columns` array.

Reply:
This deserves a Javadoc then. Although it's probably more intuitive if both indexes are absolute (less mind twisting)

Reply:
Added javadoc
getLoadedPage gets called a lot, so any non-trivial lookups start showing minor impact in JFR profiles